File: swift/socketd/socketd/src/socketd/transport/core/Typealias.swift
//
//  Typealias.swift
//  socketd
//
//  Created by noear on 2023/12/28.
//

import Foundation


typealias IoConsumer<T> = (_ t:T) -> ()
typealias IoBiConsumer<T1,T2> = (_ t1:T1,_  t2:T2 ) -> ()
typealias IoTriConsumer<T1,T2,T3> = (_ t1:T1,_  t2:T2,_ t3:T3 ) -> ()
typealias IoFunction<T1,T2> = (_ t1:T1) -> T2;

File: javascript/socketd/src/socketd/transport/core/Typealias.ts

export type IoConsumer<T> = (t:T) => void
export type IoBiConsumer<T1,T2> = (t1:T1, t2:T2 ) => void
export type IoTriConsumer<T1,T2,T3> = (t1:T1, t2:T2, t3:T3) => void
export type IoFunction<T1,T2> = (t1:T1) => T2;
export type IoSupplier<T> = ()=>T;
